programs/nodeos/main.cpp

```diff
@@ -112,9 +112,21 @@ int main(int argc, char** argv)
       return FIXED_REVERSIBLE;
    } catch( const node_management_success& e ) {
       return NODE_MANAGEMENT_SUCCESS;
+   } catch (const std_exception_wrapper& e) {
+      try {
+         std::rethrow_exception(e.get_inner_exception());
+      } catch (const std::system_error&i) {
+         if (chainbase::db_error_code::dirty == i.code().value()) {
+            elog("Database dirty flag set (likely due to unclean shutdown): replay required" );
+            return DATABASE_DIRTY;
+         }
+      } catch (...) { } 
+
+      elog( "${e}", ("e",e.what()));
+      return OTHER_FAIL;
    } catch( const fc::exception& e ) {
       if( e.code() == fc::std_exception_code ) {
-         if( e.top_message().find( "database dirty flag set" ) != std::string::npos ) {
+         if( e.top_message().find( "Database dirty flag set" ) != std::string::npos ) {
```

Comment:
You should add a catch of std::system_error and check error code instead of searching for string.

Reply:
This is a relic at this point and should be removed.